### DISCUSSION TITLE
`yes | command` can not be terminated on macOS

### DISCUSSION THREAD

**Discussion by JounQin** [community/community#26226] [Actions] [answered]
<p><a href="https://github.com/JounQin/.dotfiles/blob/master/.github/workflows/ci.yml#L16" rel="nofollow noopener">https://github.com/JounQin/.dotfiles/blob/master/.github/workflows/ci.yml#L16</a></p>
<p><a href="https://github.com/JounQin/.dotfiles/runs/438808113?check_suite_focus=true" rel="nofollow noopener">https://github.com/JounQin/.dotfiles/runs/438808113?check_suite_focus=true</a></p>
<p>I tried <code>exit 0</code> manually, doesn’t work.</p>
<p>And there’s no timeout support?</p>

**Comment by weide-zhou** [comment]
<p>I can repro the same with ‘yes | command’ for macOS runner. Please remove ‘yes’ for your workflow, it passed then.</p>
<p>If you’d like to set timeout for a job or a step, please check the usage of ‘<a href="https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idtimeout-minutes" rel="nofollow noopener">jobs.&lt;job_id&gt;.timeout-minutes</a>’ and ‘<a href="https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstepstimeout-minutes" rel="nofollow noopener">jobs.&lt;job_id&gt;.steps.timeout-minutes</a>’. Thanks.</p>
<pre><code>- name: Setup on macOS
        if: matrix.os == 'macOS-latest'
        run: sh setup.sh
        env:
          CI: true
</code></pre>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/3/396e095ce0513a7c24439a07c5dd56e970d19414.png" data-download-href="/uploads/short-url/8c2Y1kd9U84vO1614NPig8Z4adm.png?dl=1" title="macpass.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/52772761/181096151-7bbe8b0f-9b3e-466c-b1fc-d315183f8fe3.png" alt="macpass.png" data-base62-sha1="8c2Y1kd9U84vO1614NPig8Z4adm" width="690" height="226" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/3/396e095ce0513a7c24439a07c5dd56e970d19414_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">macpass.png</span><span class="informations">725×238 10.5 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by tavianator** [comment, marked as answer]
See https://github.com/actions/runner/issues/884
